src/cspviolationdetector/DefaultCSPViolationDetector.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import CSPViolationDetector from './CSPViolationDetector';
+
+export default class DefaultCSPViolationDetector implements CSPViolationDetector {
+  private static count: number = 0;
+
+  /* istanbul ignore next */
+  private static start(): void {
+    if (!('document' in global) || !document.addEventListener) {
+      return;
+    }
+    if (DefaultCSPViolationDetector.count === 0) {
+      document.addEventListener('securitypolicyviolation', DefaultCSPViolationDetector.listener);
+    }
+    DefaultCSPViolationDetector.count += 1;
+  }
+
+  /* istanbul ignore next */
+  private static stop(): void {
+    if (!('document' in global) || !document.addEventListener) {
+      return;
+    }
+    if (DefaultCSPViolationDetector.count > 0) {
+      DefaultCSPViolationDetector.count -= 1;
+      if (DefaultCSPViolationDetector.count === 0) {
+        document.removeEventListener(
+          'securitypolicyviolation',
+          DefaultCSPViolationDetector.listener
+        );
+      }
+    }
+  }
+
+  /* istanbul ignore next */
+  private static listener = (event: SecurityPolicyViolationEvent): void => {
+    console.error(
```

Comment:
We want to log to the console *as well* as _optionally_ a logger.

Why optionally? Because this log line leaks a ton of data about the application, and we don't want to force builders into sending that somewhere.

Reply:
Will change.